test/Microsoft.Azure.Devices.Edge.Test.Common/Linux/EdgeDaemon.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            var properties = new object[] { };
+            string message = "Installed edge daemon";
+            packagesPath.ForEach(
+                p =>
+                {
+                    message += " from packages in '{InstallPackagePath}'";
+                    properties = new object[] { p };
+                });
+
+            string[] commands = await packagesPath.Match(
+                p =>
+                {
+                    string[] packages = Directory.GetFiles(p, "*.deb");
+                    return Task.FromResult(
+                        new[]
+                        {
+                            "set -e",
+                            $"dpkg --force-confnew -i {string.Join(' ', packages)}",
+                            "apt-get install -f"
+                        });
+                },
+                async () =>
+                {
+                    string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
+                    string os = platformInfo[0].Trim();
+                    string version = platformInfo[1].Trim();
+                    switch (os)
+                    {
+                        case "Ubuntu":
+                            return new[]
+                            {
+                                "set -e",
+                                $"curl https://packages.microsoft.com/config/ubuntu/{version}/prod.list > /etc/apt/sources.list.d/microsoft-prod.list",
+                                "curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg",
+                                "apt-get update",
+                                "apt-get install --yes iotedge"
+                            };
+                        case "Raspbian":
+                            return new[]
+                            {
+                                "set -e",
+                                "curl -L https://aka.ms/libiothsm-std-linux-armhf-latest -o libiothsm-std.deb",
+                                "curl -L https://aka.ms/iotedged-linux-armhf-latest -o iotedge.deb",
+                                "dpkg --force-confnew -i libiothsm-std.deb iotedge.deb",
+                                "apt-get install -f",
+                                "rm libiothsm-std.deb iotedge.deb"
+                            };
+                        default:
+                            throw new NotImplementedException($"Don't know how to install daemon on operating system '{os}'");
+                    }
+                });
+
+            await Profiler.Run(
+                async () =>
+                {
+                    string[] output = await Process.RunAsync("bash", $"-c \"{string.Join("; ", commands)}\"", token);
+                    Log.Verbose(string.Join("\n", output));
+                },
+                message,
+                properties);
+
```

Comment:
nit: suggest to break down this method into 2 smaller internal methods - code above and below this comment.

Reply:
Sure I'll do that

Reply:
Done.